etl/steps/data/garden/hmd/2024-11-19/hfd.py

```diff
@@ -0,0 +1,621 @@
+    """
+    # Outer join
+    tb = tb.merge(
+        tb_bo,
+        on=col_index,
+        suffixes=["", "__bo"],
+        how="outer",
+    )
+
+    # Integrate core indicators
+    # It can happen that one table has more values for the core indicator. We solve this with fillna calls.
+    for col in core_indicators:
+        # Check that we can integrate them!
+        TOLERANCE = 5e-3
+        assert (
+            (((tb[col] - tb[f"{col}__bo"]) / tb[col]).dropna().abs() < TOLERANCE).all()
+        ).all(), f"Integration failed for {col}. Core indicator is not equivalent between main and `bo` tables."
+
+        # Actual integration
+        tb[col] = tb[col].fillna(tb[f"{col}__bo"])
+        tb = tb.drop(columns=[f"{col}__bo"])
+
+    return tb
+
+
+def make_table_with_birth_order(tb, col_index, core_indicators, col_bo="birth_order"):
+    """Change the format of a table from wide to long, to incorporate the birth order as a dimension."""
+
+    def _generate_regex(name):
+        if re.search(r"\d$", name):  # Check if the name ends with a number
+            return rf"^{name}_?(\d+|(\d+p)?)$"
+        else:
+            return rf"^{name}(\d+|(\d+p)?)$"
+
+    regex_patterns = {name: _generate_regex(name) for name in core_indicators}
+
+    tb = tb.melt(
+        col_index,
+        var_name=COLUMN_RAW,
+        value_name="value",
+    )
+
+    tb["indicator_name"] = None
+    tb[col_bo] = None
+    for name, pattern in regex_patterns.items():
+        # print(f"> {name}")
+
+        # Set indicator name
+        flag_0 = (~tb[COLUMN_RAW].isin(core_indicators)) | (tb[COLUMN_RAW] == name)
+        flag = tb[COLUMN_RAW].str.match(pattern) & flag_0
+        assert tb.loc[flag, COLUMN_IND].isna().all(), "Multiple columns assign to the same indicator!"
+        tb.loc[flag, COLUMN_IND] = name
+
+        # Get birth order
+        tb.loc[flag, col_bo] = tb.loc[flag, COLUMN_RAW].replace({f"{name}_?": ""}, regex=True)
+        tb.loc[tb[COLUMN_RAW] == name, col_bo] = "total"
+
+    # Sanity check
+    assert tb[COLUMN_IND].notna().all(), "Some NaNs found in column `indicator_name`"
+    assert tb[col_bo].notna().all(), f"Some NaNs found in column `{col_bo}`"
+
+    # Final reshape
+    tb = tb.drop(columns=[COLUMN_RAW])
+    tb = tb.pivot(index=col_index + [col_bo], columns=COLUMN_IND, values="value").reset_index()
+    tb = tb.rename_axis(None, axis=1)
+
+    # Drop NaNs
+    tb = tb.dropna(subset=core_indicators)
+
+    return tb
+
+
+def read_table(ds_meadow, tname, tname_base=None):
+    """Read table from dataset and minor cleaning:
+
+    - Rename columns if applicable
+    - Harmonize country names
+    """
+    # Read table
+    tb = ds_meadow.read(tname)
+
+    # Rename columns
+    if tname_base is None:
+        tname_base = tname
+    if tname_base in COLUMNS_RENAME:
+        tb = tb.rename(columns=COLUMNS_RENAME[tname_base])
+
+    # Harmonize country names
+    tb = geo.harmonize_countries(
```

Comment:
You could add warn_on_unused_countries = False to avoid verbosity in the terminal

Reply:
that makes a lot of sense, thanks!